src/richie/apps/core/models.py

```diff
@@ -144,3 +146,79 @@ def check_publication(self, language=None):
         """
         language = language or translation.get_language()
         return self.extended_object.is_published(language)
+
+    # pylint: disable=too-many-locals
+    def get_related_page_extensions(
+        self, model_name, language=None, include_descendants=False
+    ):
+        """
+        Return a query to get the page extentions of a given model type related to this category
```

Comment:
Copy/pasta failed?

Reply:
Yes. I turned it into an example as it is very hard to explain clearly...